opendevin/core/config.py

```diff
@@ -180,7 +181,9 @@ class AppConfig(metaclass=Singleton):
         else ':main'
     )
     run_as_devin: bool = True
-    max_iterations: int = 100
+    max_iterations_per_task: int = 100
+    max_iterations: int = 1000
+    global_max_iterations: int = 1000
```

Comment:
There are defaults, but I don't see any value sanity checks for config values coming in later, like are they > 0 or the global max iteration should be > max iteration etc.?
Should/could a 0 in either value be allowed to disable this whole limitation (in the overall PR that is)?

Reply:
wait why do we have `global_max_iterations`? I thought `max_iterations_per_task` and `max_iterations` would be enough.

Reply:
Hmm see above comment: https://github.com/OpenDevin/OpenDevin/pull/2296#discussion_r1639301360 
I thought because of the PR title it was supposed to be in?

Reply:
> because of the PR title it was supposed to be in

In #2121 I proposed we introduce a new GLOBAL_MAX_ITERATIONS, but later in this PR [here](https://github.com/OpenDevin/OpenDevin/pull/2296#discussion_r1630624378) we agreed to change `global_max_iterations` to `max_iterations_per_task`.

Reply:
welp, more work for @prithvi2226 then 😂

Reply:
haha bring it on 🦾